Userland/Libraries/LibCore/MappedFile.cpp

```diff
@@ -20,6 +21,24 @@ ErrorOr<NonnullRefPtr<MappedFile>> MappedFile::map(StringView path)
     return map_from_fd_and_close(fd, path);
 }
 
+ErrorOr<NonnullRefPtr<MappedFile>> MappedFile::map_from_file(NonnullOwnPtr<Core::File> stream, StringView path)
+{
+    auto const fd = stream->leak_fd(Badge<MappedFile> {});
+
+    TRY(Core::System::fcntl(fd, F_SETFD, FD_CLOEXEC));
+
+    ScopeGuard fd_close_guard = [fd] {
+        close(fd);
+    };
+
+    auto stat = TRY(Core::System::fstat(fd));
+    auto size = stat.st_size;
+
+    auto* ptr = TRY(Core::System::mmap(nullptr, size, PROT_READ, MAP_SHARED, fd, 0, 0, path));
+
+    return adopt_ref(*new MappedFile(ptr, size));
```

Comment:
All of this is identical to `map_from_fd_and_close()`, could we just call that?

```suggestion
    return map_from_fd_and_close(fd, path);
```

Reply:
My bad, that was really dumb of me